search/signals.py

```diff
@@ -12,26 +12,75 @@
     Employment,
     Profile,
 )
+from dashboard.models import (
+    CachedCertificate,
+    CachedEnrollment,
+)
 from search.tasks import index_users, remove_user
 
 log = logging.getLogger(__name__)
 
 
-@receiver(post_save)
-def handle_profile_update(sender, **kwargs):  # pylint: disable=unused-argument
-    """Update index when a Profile, Education, or Employment is updated."""
-    instance = kwargs["instance"]
-    if isinstance(instance, Profile):
-        index_users.delay([instance.user])
-    elif isinstance(instance, (Education, Employment)):
-        index_users.delay([instance.profile.user])
-
-
-@receiver(post_delete)
-def handle_profile_delete(sender, **kwargs):  # pylint: disable=unused-argument
-    """Update index when a Profile, Education, or Employment is deleted."""
-    instance = kwargs['instance']
-    if isinstance(instance, Profile):
-        remove_user.delay(instance.user)
-    elif isinstance(instance, (Education, Employment)):
-        index_users.delay([instance.profile.user])
+# all the following signal handlers do basically the same.
+# The reason why there is one function per sender is
+# because each signal handler needs to be hooked to a single sender
+# otherwise it would run for any `post_save`/`post_delete` coming from any model
```

Comment:
we can probably generate these functions with less repetition using a syntax like this: `handle_<action>_<model> = receiver(...)`. i don't think that's too important though, so i won't hold up the PR for it. maybe i'll open up a separate issue